src/core/common-functions/execCommands.ts

```diff
@@ -0,0 +1,50 @@
+import { execSync } from 'child_process';
+import { env } from 'process';
+import { AppLogger } from '../../logger/appLogger.js';
+
+type StdioOption = 'inherit' | 'pipe' | 'ignore';
+type ShellOption = string | undefined; 
+interface ExecuteCommandOptions {
+    cwd?: string;
+    maxRetries?: number;
+    stdio?: StdioOption;
+    shell?: ShellOption;
+    env?:ShellOption;
+}
+
+export async function executeCommandWithRetry(
+    command: string,
+    options: ExecuteCommandOptions = {} 
+) {
+    const {
+        cwd = process.cwd(),
+        maxRetries = 3,
```

Comment:
Is it always going to be 3 retries? Can this be passed as parameter? Also, make it configurable based on system/project settings

Reply:
Yes, Now we are passing the maxretries as a parameter.